src/main/java/org/elasticsearch/common/lucene/Lucene.java

```diff
@@ -103,6 +103,17 @@ public static SegmentInfos readSegmentInfos(Directory directory) throws IOExcept
     }
 
     /**
+     * Returns the number of documents in the index referenced by this {@link SegmentInfos}
+     */
+    public static int getNumDocs(SegmentInfos info) {
+        int numDocs = 0;
+        for (SegmentCommitInfo si : info) {
+            numDocs += si.info.getDocCount() - si.getDelCount();
```

Comment:
not related to the back port, but a question - why do we deduct the del count? we typically don't do that and if so - do we want to call this getNumLiveDocs?